client/extensions/woocommerce/state/ui/orders/edits/reducer.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { merge, omit, uniqueId } from 'lodash';
+import { combineReducers } from 'state/utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_UI_ORDERS_CLEAR_EDIT,
+	WOOCOMMERCE_UI_ORDERS_EDIT,
+} from 'woocommerce/state/action-types';
+
+/**
+ * Returns the updated state after an action has been dispatched.
+ * The state reflects the currently edited order ID (which might
+ * be a placeholder object).
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function currentlyEditingId( state = null, action ) {
+	switch ( action.type ) {
+		case WOOCOMMERCE_UI_ORDERS_EDIT:
+			if ( action.order && action.order.id ) {
+				return action.order.id;
+			}
+			return { placeholder: uniqueId( 'order_' ) };
```

Comment:
If I make two edits to a not-yet-persisted-to-the-server order, won't this placeholder ID roll? That seems bad. Maybe you need a conditional to test for placeholder before rolling it?

Reply:
As soon as a change is made to the order it gets its placeholder ID, so following updates will pass back the ID (as of https://github.com/Automattic/wp-calypso/pull/17966/commits/6b5c74b690c2ccc5b5b27f0628cad3b88ac213c5, in the selectors PR #17966), and it shouldn't update.